pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -681,6 +696,31 @@ func (asw *actualStateOfWorld) PodExistsInVolume(
 	return podExists, volumeObj.devicePath, nil
 }
 
+func (asw *actualStateOfWorld) PodRemovedFromVolume(
+	podName volumetypes.UniquePodName,
+	volumeName v1.UniqueVolumeName) bool {
+	asw.RLock()
+	defer asw.RUnlock()
+
+	volumeObj, volumeExists := asw.attachedVolumes[volumeName]
+	if !volumeExists {
+		return true
+	}
+
+	podObj, podExists := volumeObj.mountedPods[podName]
+	if podExists {
+		// if volume mount was uncertain we should keep trying to unmount the volume
+		if podObj.volumeMountStateForPod == operationexecutor.VolumeMountUncertain {
+			return false
+		}
+		if podObj.volumeMountStateForPod == operationexecutor.VolumeMounted {
```

Comment:
reading this again - so basically we are saying if a volume is uncertain or mounted - then it is *not* removed from pod in the kubelet and hence can be removed from DSOW.  

Previously we prevented removal of uncertain volumes from DSOW, but since we not stop deletion of pods with uncertain volumes, may be it is fine.